apps/meteor/client/hooks/account/useSavePreferences.ts

```diff
@@ -0,0 +1,57 @@
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useMutation } from '@tanstack/react-query';
+import type { UseFormReset } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+
+import { getDirtyFields } from '../../lib/getDirtyFields';
+import { dispatchToastMessage } from '../../lib/toast';
+import type { AccountPreferencesData } from '../../views/account/preferences/useAccountPreferencesValues';
+
+type useSavePreferencesProps = {
+	dirtyFields: Partial<Record<keyof AccountPreferencesData, boolean | boolean[]>>;
+	reset: UseFormReset<AccountPreferencesData>;
+	currentData: AccountPreferencesData;
+};
+
+export const useSavePreferences = ({ dirtyFields, currentData, reset }: useSavePreferencesProps) => {
```

Comment:
I would advise against refactoring the logic into a separate hook in a fix, unless it would be used in another component. Makes it hard to understand what the root cause and fix was.